lib/supavisor/client_handler.ex

```diff
@@ -107,9 +115,16 @@ defmodule Supavisor.ClientHandler do
            Supavisor.subscribe_global(node(tenant_sup), self(), tenant, db_alias),
          ps <-
            Manager.get_parameter_status(manager) do
+      db_pid =
+        if data.mode == :session do
+          db_checkout(pool, tenant, db_alias, data.timeout)
```

Comment:
What if we call it as: 

```suggestion
          db_checkout(data.mode, pool, tenant, db_alias, data.timeout)
```

So all of the session vs transaction logic is encapsulated in helper functions at the end of the module?